recirq/time_crystals/dtctask.py

```diff
@@ -0,0 +1,254 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cirq
+import recirq
+import datetime
+import itertools
+import numpy as np
+from typing import Sequence, Optional, Dict
+import os
+
+EXPERIMENT_NAME = "time_crystals"
+DEFAULT_BASE_DIR = os.path.expanduser(f'~/cirq_results/{EXPERIMENT_NAME}')
+
+
+@recirq.json_serializable_dataclass(namespace='recirq.readout_scan',
+                                    registry=recirq.Registry,
+                                    frozen=False)
+class CompareDTCTask:
+    """ A task for managing inputs to a comparison Discrete Time Crystal experiment, comparing different options for parameters
+
+    Attributes
+        dataset_id: unique identifier for this dataset
+        qubits: chain of connected qubits available for the circuit
+        cycles: number of DTC cycles to consider for circuits
+        circuit_list: symbolic DTC circuit list
+        disorder_instances: number of disorder instances averaged over
+        options_dict: dict mapping DTCTask attribute names to options for that attribute, to take a product over
+        options_order: sequence of keys in options_dict, defining order of product over options
+    """
+
+    # Task parameters
+    dataset_id: str
+
+    # experiment parameters
+    qubits: Sequence[cirq.Qid]
+    cycles: int
+    disorder_instances: int
+    circuit_list: Sequence[cirq.Circuit]
+
+    # options to take product over
+    options_dict: Dict[str, Sequence[np.ndarray]]
+    options_order: Sequence[str]
+
+    def __init__(
+            self,
+            qubits: Sequence[cirq.Qid],
+            cycles: int,
+            disorder_instances: int,
+            options_dict: Dict[str, Sequence[np.ndarray]],
+            options_order: Optional[Sequence[str]] = None):
+
+        self.dataset_id = datetime.datetime.utcnow()
+
+        self.qubits = qubits
+        self.cycles = cycles
+        self.disorder_instances = disorder_instances
+
+        # create symbolic circuit list from qubits and cycles count
+        self.circuit_list = recirq.time_crystals.symbolic_dtc_circuit_list(qubits, cycles)
+
+        self.options_dict = options_dict
+        self.options_order = list(self.options_dict.keys()) if options_order is None else options_order
+
+        # check that the input parameters are consistent
+        assert set(self.options_order) == set(self.options_dict.keys()), 'options_order and the keys of options_dict are not the same'
+        assert not {'initial_states', 'initial_state'} <= self.options_dict.keys(), 'do not supply both initial_states and initial_state'
+
+
+    @property
+    def fn(self):
+        fn = (f'{self.dataset_id}/'
+                f'{len(self.qubits)}/'
+                f'{self.cycles}/'
+                f'{self.disorder_instances}/'
+                f'{self.options_dict}')
```

Comment:
isn't the string form of `options_dict` going to be a huge mess?

Reply:
The inputs to a single DTCTask (and the inputs to CompareDTCTask) are potentially large matrices of shape (disorder_instances, num_qubits), with many inputs randomly generated according to some distribution. 
This means the fn() functions for both are currently huge messes; already too long for filenames in many cases. 

It also means that a simplification of this won't let the filenames map inputs to datasets as in other Tasks. It might be better to only use a user-provided `dataset_id` instead of including the inputs, or to not use Tasks at all. 

Is there precedent for other Tasks using many or randomized inputs? How has it been handled there?